specification/hdinsight/data-plane/Microsoft.HDInsight/preview/2018-11-01-preview/livySpark.json

```diff
@@ -0,0 +1,1128 @@
+          "format": "int32",
+          "type": "integer"
+        },
+        "archives": {
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "queue": {
+          "type": "string"
+        },
+        "name": {
+          "type": "string"
+        },
+        "conf": {
+          "x-ms-client-name": "configuration",
+          "type": "object",
+          "additionalProperties": {
+            "type": "string"
+          }
+        },
+        "heartbeatTimeoutInSecond": {
+          "format": "int32",
+          "type": "integer"
+        }
+      }
+    },
+    "SparkJobLog": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "from": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "size": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "log": {
+          "x-ms-client-name": "logLines",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SparkJobState": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "state": {
+          "type": "string"
+        }
+      }
+    },
+    "SparkStatementCollection": {
+      "type": "object",
+      "properties": {
+        "statements": {
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SparkStatement"
+          }
+        }
+      }
+    },
+    "SparkStatement": {
+      "type": "object",
+      "properties": {
+        "id": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "code": {
+          "type": "string"
+        },
+        "state": {
+          "type": "string"
+        },
+        "output": {
+          "$ref": "#/definitions/SparkStatementOutput"
+        }
+      }
+    },
+    "SparkStatementOutput": {
+      "type": "object",
+      "properties": {
+        "status": {
+          "type": "string"
+        },
+        "execution_count": {
+          "format": "int32",
+          "type": "integer"
+        },
+        "data": {
+          "type": "object"
+        }
+      }
+    },
+    "SparkStatementRequest": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string"
+        },
+        "kind": {
+          "type": "string"
+        }
+      }
+    },
+    "SparkStatementCancellationResult": {
+      "type": "object",
+      "properties": {
+        "msg": {
+          "x-ms-client-name": "cancelMessage",
+          "type": "string"
+        }
+      }
+    }
+  },
+  "parameters": {
+    "requestedByParameter": {
+      "name": "X-Requested-By",
+      "x-ms-parameter-location": "client",
```

Comment:
Sounds surprising to put that at the client level, shouldn't this be at the operation level instead? Do we really want to customer that create its initial client to provide a "requestByParameter" ?

Reply:
Fixed it. Thanks for your comment, putting it at the operation level looks better.